README.md

```diff
@@ -6,40 +6,89 @@ This repository contains an example function, that is compatible with Speckle's
-The `Run` function takes the parsed arguments that match the 3 arguments schema described [above](#functional-anatomy).
 
-## Github Action
+The `Run` function takes 2 inputs:
+
+- `AutomationContext` -> The context of your automation, which contains all the parsed information provided by the automate service as explained [above](#anatomy-of-a-function)
+- **Optional** A `struct` representing your desired input data (see [Function Inputs](#functioninputscs---the-function-inputs))
+
+The template already contains an example implementation that will count how many objects of a particular type can be found on a version.
+
+### FunctionInputs.cs - The function inputs
+
+The definition of the user-defined inputs required for the function to work. This will also be used by the `generate-schema` CLI command to inform Automate of the required inputs a user setting up an automation will need to provide.
+
+This is just a `struct` with whatever properties you need. For now, we support basic types (string, int, double) and nested `struct`s with the same type limitations as its parent.
+
+## Publishing a function
 
 We publish a [github action](https://github.com/specklesystems/speckle-automate-github-composite-action) that bundles all the steps that are needed to publish a new version of a function. The automation requires two secret values, the `SPECKLE_FUNCTION_ID` and `SPECKLE_FUNCTION_TOKEN` both of these are provided, when you register a new function on Automate.
 
+This will:
+
+- Restore any packages and build your function
+- Generate the json schema for your inputs
+- Package your function as a Docker image
+- Register the new image as a new version in Speckle Automate
+
+Once this process has successfully finished, your function should be available and discoverable in Speckle.Automate
+
 > [!NOTE]
-> You need to register your function on Speckle Automate to get the required secret values.
+> You need to register your function on Speckle Automate to get the required secret values. For new functions, this is automatically done by our GitHub Application, so no extra user setup is needed.
 
 > [!IMPORTANT]
 > After adding the Secrets to the repository secrets on Github each commit to the `main` branch will trigger the publication of a new version.
 
+## Changing the name of your solution/project
+
+Ideally, one of the first things you'll want to do is to rename your project/solution to whatever fits your function's purpose best. If you decide to do so, please note there are several other places you may need to update the paths to match your new project structure/name.
+
+- `Dockerfile` -> line 4 contains a reference to `SpeckleAutomateDotnetExample/` folder.
+- `.github/workflows/main.yml` -> line 23 sets the working directory to be `SpeckleAutomateDotnetFolder/`
+
+Update these 2 locations too in order for the github action and docker to correctly find your newly named project.
```

Comment:
Before you dive too deeply into development, give your project a name that better suits its purpose. If you decide to rename your project or solution, remember also to update a few key areas to maintain functionality:

Dockerfile: On line 4, you'll find a reference to the SpeckleAutomateDotnetExample/ folder. Update this to match your new project name.

GitHub Actions Workflow: In .github/workflows/main.yml, line 23 sets the working directory for the GitHub Action. Update this to point to your newly named project folder.

Keeping these references up-to-date ensures that GitHub Actions and Docker can correctly find and operate your project.